src/internal/m365/collection/drive/collections_tree.go

```diff
@@ -633,15 +641,17 @@ func (c *Collections) addFileToTree(
 		// Don't add new items if the new collection has already reached it's limit.
 		// item moves and updates are generally allowed through.
 		if limiter.atContainerItemsLimit(len(parentNode.files)) || limiter.hitItemLimit(countSize.numFiles) {
-			return nil, errHitLimit
+			return nil, errHitCollectionLimit
 		}
 
-		// Skip large files that don't fit within the size limit.
-		// unlike the other checks, which see if we're already at the limit, this check
+		// Don't include large files that don't fit within the size limit.
+		// Unlike the other checks, which see if we're already at the limit, this check
 		// needs to be forward-facing to ensure we don't go far over the limit.
 		// Example case: a 1gb limit and a 25gb file.
 		if limiter.hitTotalBytesLimit(fileSize + countSize.totalBytes) {
-			return nil, errHitLimit
+			// don't return errHitLimit here; we only want to skip the
+			// current file.  We may not want to skip files after it.
```

Comment:
hmm this may require an additional check either here or in the caller. It's likely there's a subtle bug in the original implementation as well. The main concern here is that if we do meet or get close to the limit then this will always skip the file but we'll continue to enumerate items. In the worst case where all other limits are set very high, we'll enumerate everything in the drive

We can tune this behavior in another PR, but maybe something better would be something like a check of `bytesSoFar + fileSize >= bytesLimit * <some small fudge factor>` and then a check for `bytesSoFar >= bytesLimit` in another place. That would make it more likely to actually reach the limit (or at least be close to it) without forcing enumeration of all items to try to get it exactly

Reply:
patched here: https://github.com/alcionai/corso/pull/4863